parquet/src/main/java/org/apache/iceberg/parquet/ParquetVariantVisitor.java

```diff
@@ -282,4 +284,18 @@ private static boolean isBinary(Type type) {
     return type.isPrimitive()
         && type.asPrimitiveType().getPrimitiveTypeName() == PrimitiveTypeName.BINARY;
   }
+
+  protected static PhysicalType type(TimestampLogicalTypeAnnotation timestamp) {
```

Comment:
This isn't a good place for a conversion method. The visitor is a superclass of both reader and writer builders, but it isn't where Parquet conversion logic should live.